### ISSUE TITLE
Shuffled signage points not correctly handled

### ISSUE THREAD

**Issue by martomi** [martomi/chiadog#116]
**Describe the bug**

There are still cases when shuffled signage points during forks are not correctly handled.

The following sequence of events generates 61 skipped SP event.
```
2021-05-13T22:12:25.148 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 60/64: 
2021-05-13T22:12:35.081 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 61/64: 
2021-05-13T22:12:41.911 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 62/64: 
2021-05-13T22:12:49.784 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 63/64: 
2021-05-13T22:12:58.453 full_node chia.full_node.full_node: INFO     ⏲️  Finished sub slot, SP 64/64, 
2021-05-13T22:13:07.659 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 1/64: 
2021-05-13T22:13:14.053 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 2/64: 
2021-05-13T22:13:42.341 full_node chia.full_node.full_node: INFO     ⏲️  Finished sub slot, SP 64/64, 
2021-05-13T22:13:50.884 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 1/64: 
2021-05-13T22:13:58.247 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 2/64: 
2021-05-13T22:14:06.288 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 3/64: 
2021-05-13T22:14:14.181 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 4/64: 
2021-05-13T22:14:22.166 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 5/64: 
2021-05-13T22:14:30.104 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 6/64: 
2021-05-13T22:14:38.398 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 7/64: 
```

One way to fix would be comparing expected time for 61 SP (61 times 8-9 seconds) with actual time difference to previous SP.

**Environment:**
 - Chia version: 1.1.5

**Comment by martomi** [comment]
Meanwhile, if anyone wants to disable these notifications, they can replace the following line with `return None`

https://github.com/martomi/chiadog/blob/889e33c13dc9ffcd8d15972c65f00b162ed70bf0/src/chia_log/handlers/condition_checkers/non_skipped_signage_points.py#L75

**Comment by St3ffn** [comment]
@martomi I really like chiabot. First usable alerting regarding chia. So thanks for the work 🎇. 

In my setup we have one remote farmer and several distrubuted harvesters. On all of them we use chiabot. What really drives me crazy is the notification about "Experiencing networking issues? Skipped N+ signage points in the last hour". The message indicates it happened in the last hour but actually I get this message much more often than once every hour. From time to time each 10 Minutes. 

First of all I think it is a good idea to have a on and off switch for this feature. Second, if the message indicates _...in the last hour_, I would expect the messages refers to the last hour running the farmer and not some kind of random time interval. Real world example from this morning. Time of notiications: 5:29, 5:39, 5:39 (again), 5:58, 6:30, 6:34....

Currenlty there are 2 Github issues regarding the signage points problem. [1796](https://github.com/Chia-Network/chia-blockchain/issues/1796) and [3507](https://github.com/Chia-Network/chia-blockchain/issues/3507). Some assume/guess it could be caused by a low spec hardware. But as some people commented in [1796](https://github.com/Chia-Network/chia-blockchain/issues/1796) the problem also exists for people with good hardware (like i7). So in the current state of chia I really don't know if I should ignore the chiabot message or be worried :). 

Lots of prose 😉. 

**tl;dr**: Would be nice to turn this feature on and off. The message refers to an hour but is sent in some kind of random intervlal

**Comment by dukeofdrum** [comment]
So is there any indicator that the shuffled signage points reduce the chance to win? And also the „end of slot not added“

all Harvester & Full Node are on version 1.1.6dev0

i have this issue:

https://github.com/Chia-Network/chia-blockchain/issues/1796#issuecomment-840715333

also lots of those:

![3C8BDB1C-4ECE-49BF-86DE-17DCD8AC14F6](https://user-images.githubusercontent.com/84147974/118239468-1da13700-b49a-11eb-88c4-fe92336fce3f.jpeg)

**Comment by martomi** [comment]
@St3ffn there'll be a fix for that both in next version of Chia itself, as well as chiadog (tomorrow). Until then, it's indeed an annoying and useless notification because of all the false alarms.

**Comment by martomi** [comment]
Unfortunately still a `chiadog` issue - but slightly different kind than before. 
https://github.com/martomi/chiadog/issues/145

Next release will add test cases for that as well and hopefully eliminate these.

**Comment by martomi** [comment]
Scratch what I said above. I just added a test to verify the reported issue and it's passing.
There are actuals skips in my logs when I checked now. For example:

```
2021-05-20T19:53:28.925 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 62/64:
2021-05-20T19:53:38.107 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 63/64: 
2021-05-20T19:53:59.905 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 1/64: 
2021-05-20T19:54:08.305 full_node chia.full_node.full_node: INFO     ⏲️  Finished signage point 2/64:
```

**Comment by jshank** [comment]
If you are running chiadog docker,  do the following but **replace** `/mnt/media/docker-configs/chiadog/` with your local config path.

1.  Copy the signage points file out of the docker container
`docker cp chiadog:/chiadog/src/chia_log/handlers/condition_checkers/non_skipped_signage_points.py /mnt/media/docker-configs/chiadog/non_skipped_signage_points.py`

2. Edit the last line of   `non_skipped_signage_points.py` to `return None` instead of `return event`
3.  Modify your `docker-compose.yaml` file so that the container will read your modified `non_skipped_signage_points.py`
```
        volumes:
            - <other volumes here>
            - /mnt/media/docker-configs/chiadog/non_skipped_signage_points.py:/chiadog/src/chia_log/handlers/condition_checkers/non_skipped_signage_points.py
```
4. Restart the chiadog docker container
